packages/nextjs/pages/index.tsx

```diff
@@ -1,14 +1,316 @@
+                      });
+                    }}
+                    onKeyDown={e => {
+                      if (e.key === "Enter") {
+                        setState({
+                          mode: CursorMode.Chat,
+                          previousMessage: state.message,
+                          message: "",
+                        });
+                      } else if (e.key === "Escape") {
+                        setState({
+                          mode: CursorMode.Hidden,
+                        });
+                      }
+                    }}
+                    placeholder={state.previousMessage ? "" : "Say something…"}
+                    value={state.message}
+                    maxLength={50}
+                  />
+                </div>
+              </>
+            )}
+            {state.mode === CursorMode.ReactionSelector && (
+              <ReactionSelector
+                setReaction={reaction => {
+                  setReaction(reaction);
+                }}
+              />
+            )}
+            {state.mode === CursorMode.Reaction && (
+              <div className="absolute top-3.5 left-1 pointer-events-none select-none">{state.reaction}</div>
+            )}
+          </div>
+        )}
+
+        {others.map(({ connectionId, presence }) => {
+          if (presence == null || !presence.cursor) {
+            return null;
+          }
+          return (
+            <>
+              <Cursor
+                key={connectionId}
+                color={COLORS[connectionId % COLORS.length]}
+                x={presence.cursor.x}
+                y={presence.cursor.y}
+                message={presence.message}
+              />
+            </>
+          );
+        })}
+        {children}
+      </div>
     </>
   );
 };
 
-export default Home;
+export default function Home() {
+  const roomId = useOverrideRoomId("nextjs-live-cursors-chat");
+
+  return (
+    <>
+      <MetaHeader />
+      <RoomProvider
+        id={roomId}
+        initialPresence={() => ({
+          cursor: null,
+          message: "",
+        })}
+      >
+        <LiveCursorContainer>
+          <CanvasComponent />
+        </LiveCursorContainer>
+      </RoomProvider>
+    </>
+  );
+}
+
+export async function getStaticProps() {
+  const API_KEY = process.env.NEXT_PUBLIC_LIVEBLOCKS_PUBLIC_KEY;
+  const API_KEY_WARNING = process.env.CODESANDBOX_SSE
+    ? `Add your public key from https://liveblocks.io/dashboard/apikeys as the \`NEXT_PUBLIC_LIVEBLOCKS_PUBLIC_KEY\` secret in CodeSandbox.\n` +
+      `Learn more: https://github.com/liveblocks/liveblocks/tree/main/examples/nextjs-live-cursors-chat#codesandbox.`
+    : `Create an \`.env.local\` file and add your public key from https://liveblocks.io/dashboard/apikeys as the \`NEXT_PUBLIC_LIVEBLOCKS_PUBLIC_KEY\` environment variable.\n` +
+      `Learn more: https://github.com/liveblocks/liveblocks/tree/main/examples/nextjs-live-cursors-chat#getting-started.`;
```

Comment:
ideally remove the warnings and add the secrets to our git repo/to the vercel deployments